s3/s3.go

```diff
@@ -0,0 +1,172 @@
+package s3
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	bindings "github.com/andream16/aws-sdk-go-bindings"
+	"github.com/andream16/aws-sdk-go-bindings/internal/format"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3iface"
+	"github.com/pkg/errors"
+)
+
+// S3er describes s3 API.
+type S3er interface {
+	CreateBucket(string) error
+	GetObject(bucket, path string) ([]byte, error)
+	PutObject(bucket, objectName, objectPath string) error
+}
+
+// S3 is the alias for s3.
+type S3 struct {
+	s3 s3iface.S3API
+}
+
+type file struct {
+	body          []byte
+	contentType   string
+	contentLength int64
+}
+
+// New returns a new S3.
+func New(region string, options ...bindings.Option) (*S3, error) {
+
+	if region == "" {
+		return nil, errors.New("required parameter region cannot be empty")
+	}
+
+	cfg := &bindings.Config{
+		Region: format.StrToPtr(region),
+	}
+
+	for _, option := range options {
+		option(cfg)
+	}
+
+	sess, err := session.NewSession((*aws.Config)(cfg))
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to initialize s3 session")
+	}
+
+	return &S3{
+		s3: s3.New(sess),
+	}, nil
+
+}
+
+// CreateBucket creates an s3 bucket.
+func (s S3) CreateBucket(bucket string) error {
+
+	if bucket == "" {
+		return errors.Wrap(bindings.ErrInvalidParameter, "bucket")
+	}
+
+	in := &s3.CreateBucketInput{
+		Bucket: format.StrToPtr(bucket),
```

Comment:
Minor: for this I'd probably take the address of the string value type: `Bucket: &bucket` - it's more explicit - plus, is getting the address of a variable associated with formatting? That is to say `&bucket` reads more obvious than a bespoke sub-package curiously named "format". You're importing AWS SDK anyway so I'd probably use `aws.String()` if you're intent on using helper functions, because most users of Go + AWS will be aware of its existence.